tools/windows.py

```diff
@@ -45,6 +46,18 @@ def generate(env):
         env["SHLIBPREFIX"] = ""
         # Want dll suffix
         env["SHLIBSUFFIX"] = ".dll"
+
+        env.Append(CCFLAGS=["-Wwrite-strings"])
+        if env["use_static_cpp"]:
+            env.Append(
+                LINKFLAGS=[
+                    "-static",
+                    "-Wl,--no-undefined",
```

Comment:
"no undefined" probably should be there for both static and dynamic.

Reply:
Makes sense! I moved it up to apply in both cases